src/Workspaces/Core/Portable/NamingStyles/Serialization/SymbolSpecification.cs

```diff
@@ -280,48 +286,66 @@ private interface ISymbolMatcher
             bool MatchesSymbol(ISymbol symbol);
         }
 
-        public struct SymbolKindOrTypeKind : IEquatable<SymbolKindOrTypeKind>, ISymbolMatcher
+        public struct SymbolOrTypeOrMethodKind : IEquatable<SymbolOrTypeOrMethodKind>, ISymbolMatcher
         {
             public SymbolKind? SymbolKind { get; }
             public TypeKind? TypeKind { get; }
+            public MethodKind? MethodKind { get; }
 
-            public SymbolKindOrTypeKind(SymbolKind symbolKind) : this()
+            public SymbolOrTypeOrMethodKind(SymbolKind symbolKind) : this()
             {
                 SymbolKind = symbolKind;
                 TypeKind = null;
+                MethodKind = null;
             }
 
-            public SymbolKindOrTypeKind(TypeKind typeKind) : this()
+            public SymbolOrTypeOrMethodKind(TypeKind typeKind) : this()
             {
                 SymbolKind = null;
                 TypeKind = typeKind;
+                MethodKind = null;
+            }
+
+            public SymbolOrTypeOrMethodKind(MethodKind methodKind) : this()
+            {
+                SymbolKind = null;
+                TypeKind = null;
+                MethodKind = methodKind;
             }
 
             public bool MatchesSymbol(ISymbol symbol)
-                => SymbolKind.HasValue
-                    ? symbol.IsKind(SymbolKind.Value)
-                    : symbol is ITypeSymbol s && s.TypeKind == TypeKind.Value;
+                => SymbolKind.HasValue ? symbol.IsKind(SymbolKind.Value) :
+                   TypeKind.HasValue ? symbol is ITypeSymbol type && type.TypeKind == TypeKind.Value :
+                   MethodKind.HasValue ? symbol is IMethodSymbol method && method.MethodKind == MethodKind.Value :
+                   throw ExceptionUtilities.Unreachable;
 
             internal XElement CreateXElement()
-                => SymbolKind.HasValue
-                     ? new XElement(nameof(SymbolKind), SymbolKind)
-                     : new XElement(nameof(TypeKind), TypeKind);
+                => SymbolKind.HasValue ? new XElement(nameof(SymbolKind), SymbolKind) :
+                   TypeKind.HasValue ? new XElement(nameof(TypeKind), TypeKind) :
+                   MethodKind.HasValue ? new XElement(nameof(MethodKind), MethodKind) :
```

Comment:
what's the impact in the other direction.  will previous versions of VS properly ignore this and not crash if you use this and have preferences-roaming on?

Reply:
they should ignore it, the code in `GetSymbolKindListFromXElement` looks for names it knows, but I can't confirm... I think we need to upgrade the options

Reply:
previous versions will notice that the version is different than what they like and they will reset the options